### ISSUE TITLE
2.40.1: Error: path ... is not in cwd ...

### ISSUE THREAD

**Issue by TkDodo** [webpro/knip#341]
with the latest release (`2.40.1`), knip starts to fail with the following error:

```
Connecting the dots...
file:///my-project/node_modules/.pnpm/globby@13.2.2/node_modules/globby/ignore.js:42
		throw new Error(`Path ${fileOrDirectory} is not in cwd ${cwd}`);
		      ^

Error: Path /my-project/packages/my-package-a/src/index.ts is not in cwd my-project/packages/my-package-b
    at toRelativePath (file:///my-project/node_modules/.pnpm/globby@13.2.2/node_modules/globby/ignore.js:42:9)
    at ProjectPrincipal.isGitIgnored (file:///my-project/node_modules/.pnpm/globby@13.2.2/node_modules/globby/ignore.js:54:21)
    at file:///my-project/node_modules/.pnpm/knip@2.40.1/node_modules/knip/dist/ProjectPrincipal.js:129:44
    at Set.forEach (<anonymous>)
    at ProjectPrincipal.analyzeSourceFile (file:///my-project/node_modules/.pnpm/knip@2.40.1/node_modules/knip/dist/ProjectPrincipal.js:118:20)
    at analyzeSourceFile (file:///my-project/node_modules/.pnpm/knip@2.40.1/node_modules/knip/dist/index.js:183:66)
    at file:///my-project/node_modules/.pnpm/knip@2.40.1/node_modules/knip/dist/index.js:249:17
    at Array.forEach (<anonymous>)
    at main (file:///my-project/node_modules/.pnpm/knip@2.40.1/node_modules/knip/dist/index.js:248:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

this is in a pnpm monorepo. I replaced our project name with `my-project` in the stack trace, as well as the two packages with `my-package-a` and `my-package-b`.

I looked at the release notes but not sure what could've caused this. Note that everything runs fine with `2.40.0`.

Any hep would be appreciated 🙏

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.40.2. See [Release 2.40.2](https://github.com/webpro/knip/releases/tag/2.40.2) for release notes._

**Comment by webpro** [comment]
Sorry about that!
